File: app/src/main/java/com/example/sasham/testproject/website/WebActivity.java
package com.example.sasham.testproject.website;

import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ProgressBar;

import com.example.sasham.testproject.BaseActivity;
import com.example.sasham.testproject.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public abstract class WebActivity extends BaseActivity {

    @BindView(R.id.web_view)
    WebView webView;

    @BindView(R.id.toolbar)
    Toolbar toolbar;

    @BindView(R.id.progress)
    ProgressBar progress;

    public static final String WEB_URL_ARGS = "url";


    public static void startActivity(String url, Context context){
        Intent intent = new Intent(context, WebActivity.class);
        intent.putExtra(WebActivity.WEB_URL_ARGS, url);
        context.startActivity(intent);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_web);

        ButterKnife.bind(this);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        openWebSite();
    }

    @Override
    public boolean onSupportNavigateUp() {
        onBackPressed();
        return true;
    }

    private void openWebSite() {
        String url=getIntent().getStringExtra(WEB_URL_ARGS);
        webView.getSettings().setJavaScriptEnabled(true);
        webView.setWebViewClient(new WebViewClient(){

            @Override
            public void onPageStarted(WebView view, String url, Bitmap favicon) {
                super.onPageStarted(view, url, favicon);
                progress.setVisibility(View.VISIBLE);
            }

            @Override
            public void onPageFinished(WebView view, String url) {
                super.onPageFinished(view, url);
                progress.setVisibility(View.GONE);
            }
        });

        webView.loadUrl(url);
        setTitle(url);
    }
}

File: app/src/main/java/com/example/sasham/testproject/website/WebActivity.kt
package com.example.sasham.testproject.website

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.webkit.WebViewClient
import com.example.sasham.testproject.R
import com.example.sasham.testproject.base.BaseActivity
import kotlinx.android.synthetic.main.activity_web.*

class WebActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_web)

        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        supportActionBar!!.setDisplayShowHomeEnabled(true)
        openWebSite()
    }

    override fun onSupportNavigateUp(): Boolean {
        onBackPressed()
        return true
    }

    @SuppressLint("SetJavaScriptEnabled")
    private fun openWebSite() {
        val url = intent.getStringExtra(WEB_URL_ARGS)
        webView.settings.javaScriptEnabled = true
        webView.webViewClient = WebViewClient()
        webView.loadUrl(url)
        title = url
    }

    companion object {

        const val WEB_URL_ARGS = "url"


        fun startActivity(url: String, context: Context) {
            val intent = Intent(context, WebActivity::class.java)
            intent.putExtra(WebActivity.WEB_URL_ARGS, url)
            context.startActivity(intent)
        }
    }
}
